src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -142,62 +152,68 @@ ReportServerSideStats(
 
   // Non-ensemble model
   if (stats.composing_models_stat.empty()) {
-    const uint64_t queue_avg_us = AverageDurationInUs(stats.queue_time_ns, cnt);
-    const uint64_t compute_input_avg_us =
-        AverageDurationInUs(stats.compute_input_time_ns, infer_cnt);
-    const uint64_t compute_infer_avg_us =
-        AverageDurationInUs(stats.compute_infer_time_ns, infer_cnt);
-    const uint64_t compute_output_avg_us =
-        AverageDurationInUs(stats.compute_output_time_ns, infer_cnt);
-    const uint64_t compute_avg_us =
-        compute_input_avg_us + compute_infer_avg_us + compute_output_avg_us;
-    const uint64_t cache_hit_avg_us =
+    const uint64_t queue_time_avg_us =
+        AverageDurationInUs(stats.queue_time_ns, stats.queue_count);
+    const uint64_t compute_input_time_avg_us = AverageDurationInUs(
+        stats.compute_input_time_ns, stats.compute_input_count);
+    const uint64_t compute_infer_time_avg_us = AverageDurationInUs(
+        stats.compute_infer_time_ns, stats.compute_infer_count);
+    const uint64_t compute_output_time_avg_us = AverageDurationInUs(
+        stats.compute_output_time_ns, stats.compute_output_count);
+    const uint64_t compute_time_avg_us = compute_input_time_avg_us +
+                                         compute_infer_time_avg_us +
+                                         compute_output_time_avg_us;
+    const uint64_t cache_hit_time_avg_us =
         AverageDurationInUs(stats.cache_hit_time_ns, cache_hit_cnt);
-    const uint64_t cache_miss_avg_us =
+    const uint64_t cache_miss_time_avg_us =
         AverageDurationInUs(stats.cache_miss_time_ns, cache_miss_cnt);
     const uint64_t total_compute_time_ns = stats.compute_input_time_ns +
                                            stats.compute_infer_time_ns +
                                            stats.compute_output_time_ns;
     // Get the average of cache hits and misses across successful requests
-    const uint64_t combined_cache_compute_avg_us = AverageDurationInUs(
-        stats.cache_hit_time_ns + stats.cache_miss_time_ns +
-            total_compute_time_ns,
-        cnt);
+    const uint64_t combined_cache_compute_time_avg_us =
+        cache_miss_cnt + cache_hit_cnt > 0
+            ? ((cache_miss_cnt *
+                (cache_miss_time_avg_us + compute_time_avg_us)) +
+               (cache_hit_cnt * cache_hit_time_avg_us)) /
+                  (cache_miss_cnt + cache_hit_cnt)
+            : 0;
 
     if (parser->ResponseCacheEnabled()) {
-      const uint64_t overhead_avg_us = GetOverheadDuration(
-          cumm_avg_us, queue_avg_us, combined_cache_compute_avg_us);
+      const uint64_t overhead_time_avg_us = GetOverheadDuration(
```

Comment:
@matthewkotila As a side note, I don't think whatever issue is being fixed here is detected by CI. 

Do you think any of your debugging/testing could be added to one of the existing CI perf_analyzer tests to catch issues like this in the future?

Reply:
There was no prior testing for this, unfortunately. If it's alright, I'll do this in a future PR/ticket 👍

Reply:
Sure, a separate PR is fine - just want a way to catch similar issues automatically next time if it was reproducible.

Reply:
Created separate ticket 👍